### PR TITLE
feat: testing, comments and prepare for release

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary

- add more testing
- prep for release
- add more comments

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/experiments-python-server/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: src/amplitude_experiment/local/client.py [hunk 1]
```diff
@@ -85,10 +113,11 @@ def __setup_connection_pool(self):
 
     def close(self) -> None:
         """
-        Close resource like connection pool with client
```
Comment (yuhao900914) [inline src/amplitude_experiment/local/client.py:88]: Node using `stop`. Should we keep it aligned?

File: src/amplitude_experiment/local/client.py [hunk 2]
```diff
@@ -85,10 +113,11 @@ def __setup_connection_pool(self):
...
+        Stop polling for flag configurations. Close resource like connection pool with client
         """
         self.poller.stop()
         self._connection_pool.close()
+        self.is_running = False
```
Comment (yuhao900914) [inline src/amplitude_experiment/local/client.py:120]: why do we need is_runing at the client level?

Comment (qingzhuozhen) [reply src/amplitude_experiment/local/client.py:120]: That's to support we can call `start` multiple times. Right now the start each time will trigger a fetch, I guess still optional though, got reminded from our earlier discussion

File: src/amplitude_experiment/local/client.py (hunk 1 above)
Comment (qingzhuozhen) [reply src/amplitude_experiment/local/client.py:88]: I pick `close` earlier to align with remote client. I can also rename if we feel it is better to align with node

File: src/amplitude_experiment/local/client.py (hunk 2 above)
Comment (qingzhuozhen) [reply src/amplitude_experiment/local/client.py]: Another extra thought and would take this down, multiple start will just be multiple fetch and seems not hurting poller.

File: src/amplitude_experiment/local/client.py (hunk 1 above)
Comment (qingzhuozhen) [reply src/amplitude_experiment/local/client.py:88]: Just updated, let's try rename for now

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**